### DISCUSSION TITLE
autorun 如果在 tracker callback 里 dispose 掉？

### DISCUSSION THREAD

**Discussion by jo0ger** [alibaba/formily#3721] [Q&A 2.x]
目前场景如下，就是需要监听数据状态，然后数据状态 ready 时候，取消掉 autorun

```ts
const dispose = autorun(() => {
   const data = getData()
   if (data.status === 'ready') {
       // 想要在 数据 ready 时候 dispose 掉
       dispose()
   }
})
```

但是我看 autorun 的源码，tracer callback 是同步执行的，内部无法拿到 dispose

https://github.com/alibaba/formily/blob/formily_next/packages/reactive/src/autorun.ts#L52

**Comment by spt110** [comment]
依赖变化autorun会重复执行，首次执行是为了收集依赖。callback内逻辑，你可以自己做异步处理
